conbench/entities/commit.py

```diff
@@ -578,26 +579,111 @@ def _parse_commit(commit):
         }
 
     def _get_response(self, url) -> dict:
-        """
+        """Attempt to get HTTP response with retrying behavior towards a
+        best-effort approach in view of typical retryable errors.
+
+        Do not try for too long because there is an HTTP client waiting for
+        _us_ to generate an HTTP response in a more or less timely fashion.
+        Gunicorn has a worker timeout behavior (as of the time of writing: 30
+        seconds) and the retrying method below must come to a conclusion before
+        that.
+
         Return deserialized JSON-structure or raise an exception.
         """
-        # This can raise exceptions corresponding to transient issues related
-        # to DNS, TCP, HTTP during sending request, while waiting for response,
-        # or while receiving the response.
-        response = self.session.get(url) if self.session else requests.get(url)
-
-        # An HTTP response was received.
-        if response.status_code != 200:
-            # This is here to log details an make debugging easier.
+        timeout_seconds = 20
+
+        t0 = time.monotonic()
+        deadline = t0 + timeout_seconds
+        attempt: int = 0
+
+        while time.monotonic() < deadline:
```

Comment:
This is btw a standard recipe I have used many times over the years. In _this_ part of the code I rather have all the flexibility and also readability (better than using an involved retrying library IMO).

The only "success" way to leave this function is via `return result`. In all other cases we at some point break from this loop and raise the "deadline exceeded" exception.